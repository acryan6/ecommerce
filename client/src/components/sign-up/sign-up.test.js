import { shallow } from "enzyme";
import React from "react";
import { SignUp } from "./sign-up.component.jsx";

describe("sign-up component", () => {
  let wrapper;
  let mockSignUpStart;

  beforeEach(() => {
    mockSignUpStart = jest.fn();

    const mockProps = {
      signUpStart: mockSignUpStart,
    };

    wrapper = shallow(<SignUp {...mockProps} />);
  });

  it("expect to render sign-up component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("correctly handles submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(mockSignUpStart).toHaveBeenCalled();
  });
});
