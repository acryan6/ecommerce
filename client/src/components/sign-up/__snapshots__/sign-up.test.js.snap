// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-up component expect to render sign-up component 1`] = `ShallowWrapper {}`;
